File: identity_stardust/examples/ex2_resolve_did.rs
// Copyright 2020-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use identity_stardust::StardustClientExt;
use identity_stardust::StardustDocument;
use iota_client::block::output::AliasOutput;
use iota_client::block::output::OutputId;
use iota_client::Client;

mod ex0_create_did;

/// Demonstrate how to resolve an existing DID in an Alias Output.
#[tokio::main]
async fn main() -> anyhow::Result<()> {
  let (client, _, _, document): (Client, _, _, StardustDocument) = ex0_create_did::run().await?;

  // Obtains the Alias Output and extracts the contained document.
  let resolved_doc: StardustDocument = client.resolve_did(document.id()).await?;

  assert_eq!(resolved_doc, document);

  println!("Resolved DID Document: {resolved_doc:#?}");

  // We can also resolve the Alias Output directly.
  let (_output_id, alias_output): (OutputId, AliasOutput) = client.resolve_did_output(document.id()).await?;

  println!("The Alias Output holds {} tokens", alias_output.amount());

  Ok(())
}

File: bindings/wasm/examples-stardust/src/ex2_resolve_did.ts
// Copyright 2020-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import type {StardustDocument} from '../../node';
import type {IAliasOutput} from '@iota/iota.js';

import {createIdentity} from "./ex0_create_did";

/** Demonstrates how to resolve an existing DID in an Alias Output. */
export async function resolveIdentity() {
    // Creates a new wallet and identity (see "ex0_create_did" example).
    const {didClient, did} = await createIdentity();

    // Resolve the associated Alias Output and extract the DID document from it.
    const resolved: StardustDocument = await didClient.resolveDid(did);
    console.log("Resolved DID document:", JSON.stringify(resolved, null, 2));

    // We can also resolve the Alias Output directly.
    const aliasOutput: IAliasOutput = await didClient.resolveDidOutput(did);
    console.log("The Alias Output holds " + aliasOutput.amount + " tokens");
}
